src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/TestSuiteStepsFlow.tsx

```diff
@@ -0,0 +1,134 @@
+import {useEffect, useMemo, useState} from 'react';
+import ReactFlow, {Controls, Edge, Node} from 'reactflow';
+
+import {nanoid} from '@reduxjs/toolkit';
+
+import {notificationCall} from '@src/components/molecules';
+import {LocalStep} from '@src/models/testSuite';
+
+import IntersectionNode from './Nodes/IntersectionNode';
+import StepNode from './Nodes/StepNode';
+import {ReactFlowContainer} from './SettingsTests.styled';
+
+interface ExtendedNode extends Node {
+  group?: number | string;
+}
+
+type TestSuiteStepsFlowProps = {
+  steps: LocalStep[][];
+  setSteps: (steps: LocalStep[][]) => void;
+  showModal: (value: string, group: number | string) => void;
+};
+
+// Configure
+export const intersectionWidth = 32;
+export const intersectionHeight = 46;
+export const itemWidth = 300;
+export const itemHeight = 68;
+const horizontalGap = 150;
+const verticalGapBeforeAdd = 20;
+const verticalGapBetweenItems = 32;
+const itemVerticalSpace = verticalGapBetweenItems + 68;
+const itemHorizontalSpace = itemWidth + horizontalGap;
+
+const getIntersectionPosition = (group: number) => ({
+  x: group * itemHorizontalSpace - (horizontalGap / 2 + intersectionWidth / 2),
+  y: (itemHeight - intersectionHeight) / 2,
+});
+const getAddPosition = (group: number, groupLength: number) => ({
+  x: group * itemHorizontalSpace + (itemWidth / 2 - intersectionWidth / 2),
+  y: groupLength * itemVerticalSpace - verticalGapBeforeAdd,
+});
+const getItemPosition = (group: number, itemIndex: number) => ({
+  x: group * itemHorizontalSpace,
+  y: itemIndex * itemVerticalSpace,
+});
+
+const TestSuiteStepsFlow: React.FC<TestSuiteStepsFlowProps> = props => {
+  const {steps, setSteps, showModal} = props;
+
+  const [nodes, setNodes] = useState<ExtendedNode[]>([]);
+  const [edges, setEdges] = useState<Edge[]>([]);
+
+  const removeNode = (id: string, group: number) => {
+    if (steps[group].length === 1) {
+      if (steps[group][0].id === id) {
+        setSteps([...steps.slice(0, group), ...steps.slice(group + 1)]);
+      } else {
+        notificationCall('failed', 'Step was not removed');
```

Comment:
I'm not sure if that notification should be shown - it would be rather internal error, not user error, and may have no implications

Reply:
I think so 👍